asv_bench/benchmarks/utils/common.py

```diff
@@ -497,7 +497,7 @@ def execute(
             return
         partitions = df._query_compiler._modin_frame._partitions.flatten()
         if len(partitions) > 0 and hasattr(partitions[0], "wait"):
-            all(map(lambda partition: partition.wait(), partitions))
+            all(map(lambda partition: partition.wait() or True, partitions))
```

Comment:
I would suggest using either a simple for-loop or a special partition manager's method
```suggestion
            df._query_compiler._modin_frame._partition_mgr_cls.wait_partitions(partitions)
```
btw, the further code in this function still uses the incorrect logic

Reply:
> btw, the further code in this function still uses the incorrect logic

Where exactly?

Reply:
I would agree on using the special parallel waiting method if we always have the partition manager accessible.

Reply:
> > btw, the further code in this function still uses the incorrect logic
> 
> Where exactly?

okay, I was mistaken, the further code seems to work okay as the engine-native `wait` methods return something bool-True-like. But still, It's not clear to me why are we using `all(map())` there, IMHO it's harder to read and it takes more lines of code rather than a simple for-loop.

Reply:
I believe that `all(map())` are compiled by python into more efficient code than a simple for-loop where there may be more complex code.
Will try `wait_partitions` function.